torch/_inductor/lowering.py

```diff
@@ -1541,8 +1541,8 @@ def select_scatter(x, src, dim: int, index: int):
     def inner_fn(idx):
         return ops.where(
             ops.eq(
-                ops.index_expr(idx[dim], torch.int32),
-                ops.index_expr(index, torch.int32),
+                ops.index_expr(idx[dim] - index, torch.float),
+                ops.constant(0, torch.float),
```

Comment:
I don't think it is proper to address the issue here since there is nothing wrong that indices are int32 types. We should fix this from the cpp codegen when handling the vectorized bool masks represented as float vector. You can check out the `to_float_mask` in `cpp_prefix.h`.